tsdb/engine/tsm1/wal_test.go

```diff
@@ -701,6 +703,124 @@ func TestWALRollSegment(t *testing.T) {
 	}
 }
 
+func TestWAL_DistSize(t *testing.T) {
+	test := func(w *tsm1.WAL, oldZero, curZero bool) () {
+		// get disk size by reading file
+		files, err := ioutil.ReadDir(w.Path())
+		if err != nil {
+			t.Fatalf("read dir err: %v", err)
+		}
+
+		sort.Slice(files, func(i, j int) bool {
+			return files[i].Name() < files[j].Name()
+		})
+
+		var old, cur int64
+		for i, f := range files {
+			if i == len(files)-1 {
+				cur = f.Size()
+			} else {
+				old += f.Size()
+			}
+		}
+
+		// test zero size condition
+		if oldZero && old > 0 {
+			t.Fatalf("expect old disk size: 0, got: %d", old)
+		}
+		if !oldZero && old == 0 {
+			t.Fatalf("expect old disk size larger than 0, got 0")
+		}
+
+		if curZero && cur > 0 {
+			t.Fatalf("expect current disk size: 0, got: %d", cur)
+		}
+		if !curZero && cur == 0 {
+			t.Fatalf("expect current disk size larger than 0, got 0")
+		}
+
+		// test method DiskSizeBytes
+		if got, exp := w.DiskSizeBytes(), old+cur; got != exp {
+			t.Fatalf("expect total disk size: %d, got: %d", exp, got)
+		}
+
+		// test Statistics
+		ss := w.Statistics(nil)
+		assert.Equal(t, 1, len(ss))
+
+		m := ss[0].Values
+		assert.NotNil(t, m)
+
+		if got, exp := m["oldSegmentsDiskBytes"].(int64), old; got != exp {
+			t.Fatalf("expect old disk size: %d, got: %d", exp, got)
+		}
+
+		if got, exp := m["currentSegmentDiskBytes"].(int64), cur; got != exp {
+			t.Fatalf("expect current disk size: %d, got: %d", exp, got)
+		}
+	}
+
+	dir := MustTempDir()
+	defer os.RemoveAll(dir)
+
+	w := tsm1.NewWAL(dir)
+
+	const segSize = 1024
+	w.SegmentSize = segSize
+
+	// open
+	if err := w.Open(); err != nil {
+		t.Fatalf("error opening WAL: %v", err)
+	}
+
+	test(w, true, true)
+
+	// write without rollSegment
+	values := map[string][]tsm1.Value{
+		"cpu,host=A#!~#value": {tsm1.NewValue(1, 1.0)},
+		"cpu,host=B#!~#value": {tsm1.NewValue(1, 1.0)},
+		"cpu,host=C#!~#value": {tsm1.NewValue(1, 1.0)},
+	}
+
+	if _, err := w.WriteMulti(values); err != nil {
+		fatal(t, "write points", err)
+	}
+
+	test(w, true, false)
+
+	// write with rollSegment
```

Comment:
Not super familiar with this code, so I'm confused by this comment when I don't see `rollSegment` below. Can you clarify? Something like:
```suggestion
	// write enough points to force the segment to roll
```